packages/hardhat/scripts/generateWagmiConfig.ts

```diff
@@ -0,0 +1,63 @@
+import * as fs from "fs";
+import * as path from "path";
+const DEPLOYMENTS_PATH = "deployments";
+
+function getParsedDeploymentsForWagmiConfig() {
+  // getting all directories inside deployments directory
+  const allChainDirectories = fs.readdirSync(DEPLOYMENTS_PATH);
+  // final constructed deployments object
+  const finalObj: Record<string, Record<string | number, any>> = {};
+
+  for (const dir of allChainDirectories) {
+    const dirPath = path.join(DEPLOYMENTS_PATH, dir);
+    const filesInChainsDir = fs.readdirSync(dirPath);
+    const deployedContractNameArr = filesInChainsDir.filter(value => value.includes(".json"));
+    for (const contractFileName of deployedContractNameArr) {
+      const filePath = path.join(dirPath, contractFileName);
+      const data = fs.readFileSync(filePath, "utf8");
+      const parsedData = JSON.parse(data);
+      const chainId = fs.readFileSync(path.join(dirPath, ".chainId"), "utf8");
+      const chainToAddress = {
+        [chainId]: parsedData.address,
+      };
+      const contractName = contractFileName.split(".")[0];
+      finalObj[contractName] = { ...finalObj[contractName], ...chainToAddress };
+    }
+  }
+
+  return finalObj;
+}
+/**
+ * Generates wagmi.config.ts file with deployments address mapped with contract
+ */
+async function main() {
+  if (!fs.existsSync(DEPLOYMENTS_PATH)) {
+    console.log("No deployments found!");
+    return;
+  }
+  const result = getParsedDeploymentsForWagmiConfig();
+  fs.writeFileSync(
+    "./wagmi.config.ts",
+    `import { defineConfig } from "@wagmi/cli";
+      import { hardhat, react } from "@wagmi/cli/plugins";
+      
+      export default defineConfig({
+        out: "../nextjs/generated/contractHooks.ts",
+        contracts: [],
+        plugins: [
+          hardhat({
+            project: "./",
+            deployments: ${JSON.stringify(result, null, 2)} 
```

Comment:
This config generation feels unnecessary.   Could it just import the deployment artifacts directly in the wagmi config?

Reply:
Good catch, Thanks !! Update it at [ba65263](https://github.com/scaffold-eth/se-2/pull/200/commits/ba65263479ddb4ceece10fd6e9c79f9675afac42). 

Although had to use the `getParsedDeploymentsForWagmiConfig` function to generate [`deployments`](https://wagmi.sh/cli/plugins/hardhat#deployments-optional) options since we don't want user to pass in `contractAddress` to generated `hooks` and it should be infered internally checout -> https://github.com/scaffold-eth/se-2/pull/200#issuecomment-1440507844